.github/workflows/ci-core-reusable.yml

```diff
@@ -63,6 +63,9 @@ jobs:
       - name: Rust unit tests
         run: ci_run zk test rust
 
+      - name: Prover "smoke" tests
+        run: ci_run bash -c "cd prover && cargo check --all-targets"
```

Comment:
Why not clippy?

Reply:
Not by much; clippy is general designed to be a drop-in replacement for check that does more analysis. The principle is the same, it just has more checks.